src/benchmark_register.cc

```diff
@@ -224,12 +225,22 @@ bool FindBenchmarksInternal(const std::string& re,
   return BenchmarkFamilies::GetInstance()->FindBenchmarks(re, benchmarks, Err);
 }
 
+// This function is exclusively for generating a unique identifier for a benchmark
+// This is to uniquely identify both statistics and benchmarks runs
+// in the case where names of benchmarks have whole names and prefixes which overlap
+// when multiple benchmarks are run
+int GetBenchmarkUniqueId() {
+  static std::atomic<int> benchid(0);
+  return benchid.fetch_add(1);
```

Comment:
You can use `std::memory_order_relaxed` here.

Reply:
Thanks! I specified that memory order.